File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/integrations/IntegrationHandler.java
package me.lokka30.levelledmobs.bukkit.integrations;

import java.util.HashSet;
import java.util.Set;
import me.lokka30.levelledmobs.bukkit.integrations.internal.CitizensIntegration;
import me.lokka30.levelledmobs.bukkit.integrations.internal.RtuLangApiIntegration;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings("unused")
public final class IntegrationHandler {

    /* vars */

    private final Set<Integration> integrations = new HashSet<>();

    private final Set<Integration> internalIntegrations = Set.of(
        new CitizensIntegration(),
        new RtuLangApiIntegration()
    );

    /* constructors */

    public IntegrationHandler() {
        integrations.addAll(internalIntegrations);
    }

    /* methods */

    /* getters and setters */

    @NotNull
    public Set<Integration> getIntegrations() { return integrations; }

    @NotNull
    public Set<Integration> getInternalIntegrations() { return internalIntegrations; }

}

File: levelledmobs-plugin-bukkit/src/main/kotlin/me/lokka30/levelledmobs/bukkit/integrations/IntegrationHandler.kt
package me.lokka30.levelledmobs.bukkit.integrations

import me.lokka30.levelledmobs.bukkit.integrations.internal.CitizensIntegration
import me.lokka30.levelledmobs.bukkit.integrations.internal.RtuLangApiIntegration

class IntegrationHandler {

    val internalIntegrations = setOf(
        CitizensIntegration(),
        RtuLangApiIntegration()
    )

    val integrations = mutableSetOf<Integration>()

    init {
        integrations.addAll(internalIntegrations)
    }
}